src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,260 @@
+        Todo newTodo = new Todo(description, TaskType.TODO);
+        taskList.add(newTodo);
+        System.out.print(HORIZONTAL_LINE + "Got it! I've added this task:\n" +
+                newTodo + "\n" + HORIZONTAL_LINE);
+    }
+
+    private static void addEvent(String input, List<Task> taskList) throws DukeException {
+        String[] separated = splitDescriptionFromTiming(TaskType.EVENT, input);
+        if (separated.length == 1) {
+            throw new DukeException("Give me a timing for the event too man come on...");
+        }
+        String description = separated[0];
+        String time = separated[1];
+        Event newEvent = new Event(description, TaskType.EVENT, time);
+        taskList.add(newEvent);
+
+        System.out.print(HORIZONTAL_LINE + "Got it! I've added this task:\n" +
+                newEvent + "\n" + HORIZONTAL_LINE);
+    }
+
+    private static void addDeadline(String input, List<Task> taskList) throws DukeException {
+        String[] separated = splitDescriptionFromTiming(TaskType.DEADLINE, input);
+        if (separated.length == 1) {
+            throw new DukeException("Tell me more about the deadline too man come on...");
+        }
+        String description = separated[0];
+        String time = separated[1];
+        Deadline newDeadline = new Deadline(description, TaskType.DEADLINE,time);
+        taskList.add(newDeadline);
+
+        System.out.print(HORIZONTAL_LINE + "Got it! I've added this task:\n" +
+                newDeadline + "\n" + HORIZONTAL_LINE);
+    }
+
+    /**
+     * Deletes given tasks from the provided list of tasks. Accepts multiple tasks in one input and provides feedback
+     * if invalid inputs are detected. Can identify task numbers amidst redundant input (e.g. done 1, 2 and 3)
+     *
+     * @param userInput String of user input containing task numbers to be deleted.
+     * @param list      List of tasks
+     */
+    public static void deleteTask(String userInput, List<Task> list) throws DukeException {
+        int[] tasksToDelete = extractInt(userInput);
+
+        System.out.println(HORIZONTAL_LINE + "Awesome! I shall now try to delete the following...");
+        for (int taskNumber : tasksToDelete) {
+            // tasksToDelete may contain '0's from current implementation of extractInt method
+            if (taskNumber == 0) {
+                continue;
+            }
+            try {
+                // TODO: Check whether the given task is already done and throw an exception if it is already done
+                Task currTask = list.get(taskNumber - 1);
+                System.out.println("Deleted: " + currTask);
+                list.remove(currTask);
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println("Dude you've given be an invalid task number [" + taskNumber + "] ! Skipping...");
+            }
+        }
+        System.out.print("Boom, insect. The tasks are now DELETED!\n" + HORIZONTAL_LINE);
```

Comment:
You should return a string to be printed in your controller function instead of printing here.